tap_salesforce/client.py

```diff
@@ -255,11 +257,14 @@ def _make_request(self, method, path, data=None, params=None) -> requests.Respon
             method, url, headers=headers, params=params, data=data
         )
 
-        json = resp.json()
-        if isinstance(json, list) and isinstance(json[0], dict) and json[0]['errorCode'] == 'QUERY_TIMEOUT':
-            raise QueryTimeoutError()
-
-        resp.raise_for_status()
+        if resp.status_code < 200 or resp.status_code > 299:
```

Comment:
In case of redirection not handled, we will fallback to `resp.raise_for_status()`. That will end up with a `raise HTTPError('', self)`.

It's more clear to see this in last commit - 9dcdc2c26d2607e6224afe5da862a41a0fe6d242 (added just after you wrote your comments!).

Reply:
Oh no, let me swallow my words. It only takes range [400, 600)

Reply:
If we do know that is handled by the framework, do we want to pollute the code with a checking we know it should never happen?

Reply:
Don't bother checking for it imo, it's either handled by `requests` or it's not, in which case we'll get a `KeyError` at some point.